### PR TITLE
Permit multiple indirections in coding calldata to and from memory/calldata

### PR DESCRIPTION
**Pull request by bshastry**
Generalizes the number of indirections used to test calldata-to-calldata/calldata-to-memory and vice versa.

### CODE REVIEW

**Review by ekpyron — COMMENTED** [review]
One question to make it quicker to understand this:

Will each indirection independently choose whether each argument is in ``calldata`` or ``memory``? (I.e. independently of the other indirection-function and independently of the other arguments in the same indirection-function?)

**Comment by bshastry** [comment]
> One question to make it quicker to understand this:
> 
> Will each indirection independently choose whether each argument is in `calldata` or `memory`? (I.e. independently of the other indirection-function and independently of the other arguments in the same indirection-function?)

Yes, iiuc.

Example 1: https://gist.github.com/bshastry/5fb0f7262c11f3ee866b83b918c7205c
Example 2: https://gist.github.com/bshastry/622366ba9b962c4629aa9f4dd2167990

File: test/tools/ossfuzz/protoToAbiV2.h [hunk 1]
```diff
@@ -418,10 +440,16 @@ class ProtoConverter
...
+	/// Representation of external parameters
+	std::vector<ParameterPack> m_externalParamsRep;
+	/// Random number generator
+	std::unique_ptr<RandomEngine> m_random;
 	/// Prefixes for declared and parameterized variable names
 	static auto constexpr s_localVarNamePrefix = "lv_";
 	static auto constexpr s_stateVarNamePrefix = "sv_";
 	static auto constexpr s_paramNamePrefix = "p_";
+	/// Maximum number of indirections to test calldata coding
+	static unsigned constexpr s_maxIndirections = 10;
```
Comment (ekpyron) [inline test/tools/ossfuzz/protoToAbiV2.h]: I guess
```suggestion
	static unsigned constexpr s_maxIndirections = 5;
```
would also be enough :-).

**Review by ekpyron — DISMISSED** [review]
Nice :-)!

File: test/tools/ossfuzz/protoToAbiV2.h (hunk 1 above)
Comment (bshastry) [reply test/tools/ossfuzz/protoToAbiV2.h]: Sounds good! Done!

**Review by ekpyron — APPROVED** [review]

**Pull request merged**